From 0878d933223158343af3fe13e03bcd128e8b4cbc Mon Sep 17 00:00:00 2001
From: Fabrice Fontaine <fabrice.fontaine@orange.com>
Date: Fri, 22 Apr 2016 17:06:27 +0200
Subject: [PATCH 1/1] Fix warning on wcsdup implicit declaration

Define __GNU_SOURCE in cpp/hidapi/linux/hid.c (like already done in
cpp/hidapi/libusb/hid.c) to fix implicit declaration of wcsdup function

Signed-off-by: Fabrice Fontaine <fabrice.fontaine@orange.com>
---
 cpp/hidapi/linux/hid.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/cpp/hidapi/linux/hid.c b/cpp/hidapi/linux/hid.c
index dbf9b9b..c3eb22b 100644
--- a/cpp/hidapi/linux/hid.c
+++ b/cpp/hidapi/linux/hid.c
@@ -21,6 +21,8 @@
         http://github.com/signal11/hidapi .
 ********************************************************/
 
+#define _GNU_SOURCE /* needed for wcsdup() before glibc 2.10 */
+
 /* C */
 #include <stdio.h>
 #include <string.h>
-- 
1.9.1

